src/e2e-test/java/io/cdap/plugin/common/ValidationHelper.java

```diff
@@ -0,0 +1,278 @@
+    Page<Blob> blobs = storage.list(bucketName);
+
+    List<Blob> bucketObjects = StreamSupport.stream(blobs.iterateAll().spliterator(), true)
+      .filter(blob -> blob.getSize() != 0)
+      .collect(Collectors.toList());
+
+    Stream<String> objectNamesWithData = bucketObjects.stream().map(blob -> blob.getName());
+    List<String> bucketObjectNames = objectNamesWithData.collect(Collectors.toList());
+
+    if (!bucketObjectNames.isEmpty()) {
+      String objectName = bucketObjectNames.get(0);
+      if (objectName.contains("part-r")) {
+        Map<String, JsonObject> dataMap2 = fetchObjectData(PROJECT_ID, bucketName, objectName);
+        dataMap.putAll(dataMap2);
+      }
+    }
+
+    return dataMap;
+  }
+
+  private static Map<String, JsonObject> fetchObjectData(String projectId, String bucketName, String objectName) {
+    Map<String, JsonObject> dataMap = new HashMap<>();
+    Storage storage = StorageOptions.newBuilder().setProjectId(projectId).build().getService();
+    byte[] objectData = storage.readAllBytes(bucketName, objectName);
+    String objectDataAsString = new String(objectData, StandardCharsets.UTF_8);
+
+    // Splitting using the delimiter as a File can have more than one record.
+    String[] lines = objectDataAsString.split("\n");
+    for (String line : lines) {
+      JsonObject json = gson.fromJson(line, JsonObject.class);
+      String id = json.get("transaction_uid").getAsString();
+      dataMap.put(id, json);
+    }
+
+    return dataMap;
+  }
+
+  private static void getBigQueryTableData(String table, Map<String, JsonObject> bigQueryRows)
+    throws IOException, InterruptedException {
+
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+    for (FieldValueList row : result.iterateAll()) {
+      JsonObject json = gson.fromJson(row.get(0).getStringValue(), JsonObject.class);
+      JsonElement idElement = json.get("transaction_uid");
+      if (idElement != null && idElement.isJsonPrimitive()) {
+        String id = idElement.getAsString(); // Replace "id" with the actual key in the JSON
+        bigQueryRows.put(id, json);
+      } else {
+        System.out.println("Data Mismatched");
+      }
+    }
+  }
+
+  private static boolean matchJsonMaps(Map<String, JsonObject> map1, Map<String, JsonObject> map2) {
+    if (!map1.keySet().equals(map2.keySet())) {
+      return false;
+    }
+    for (String key : map1.keySet()) {
+      JsonObject json1 = map1.get(key);
+      JsonObject json2 = map2.get(key);
+      if (!validateJsonElement(json1, json2)) {
+        return false;
+      }
+    } return true;
+  }
+
+  public static boolean validateJsonArray(JsonArray v1, JsonArray v2) {
```

Comment:
When you are fetching data from GCS or BQ they must be returning Json. Where are you getting these types from ?

Reply:
When we are fetching the data in the form JSON. It comes in the format of JSON Elements. (JsonPrimitive, JsonArray, JsonObject) which are nested to each other. So based on JSON Element, Validation has been done.